pkg/controller/certificates/signer/signer.go

```diff
@@ -111,13 +94,65 @@ func (s *signer) handle(csr *capi.CertificateSigningRequest) error {
 	return nil
 }
 
+// maybeReloadCA stores the current cert/key content if the values have changed
+func (s *signer) maybeReloadCA() error {
```

Comment:
Why not just call this at the top of currentCA?

Reply:
> Why not just call this at the top of currentCA?

Because it calls `currentCA` to determine if needs replacing.

Reply:
Something like:

```
func (s *signer) currentCA() *authority.CertificateAuthority {
  certPEM, keyPEM := s.caLoader.CurrentCertKeyContent()
  currCA := s.caValue.Load().(*authority.CertificateAuthority)
  if bytes.Equal(currCA.RawCert, certPEM) && bytes.Equal(currCA.RawKey, keyPEM) {
    return nil
  }

  return s.setCA(s.caLoader)
}
```

Given that this will should always be called before currentCA (or currentCA may not correctly return the currentCA). This aligns with the contract of caLoader.CurrentCertKeyContent.